torchdynamo/eval_frame.py

```diff
@@ -228,18 +228,73 @@ def toy_example(a, b):
         backend_ctx_ctor = getattr(backend, "backend_ctx_ctor")
 
     if nopython:
-        return optimize_assert(backend, backend_ctx_ctor)
+        return optimize_assert(backend, backend_ctx_ctor, guard_export_fn=None)
     return _optimize_catch_errors(
-        convert_frame.convert_frame(backend), backend_ctx_ctor
+        convert_frame.convert_frame(backend, guard_export_fn=None), backend_ctx_ctor
     )
 
 
-def optimize_assert(backend, backend_ctx_ctor=null_context):
+def export(f, *args, **kwargs):
```

Comment:
From @jansel, copying as it got lost due to renames:

I'd like to see some checks in this function that the inputs/outputs match.

One issue is if the user gives a function that doesn't take a flat list of tensors:

If the user does:

foo(a, [b,c], {"d": d})
The graph will just take foo_graph(a,b,c,d).

You have the same issue for return values. We could use pytrees to flatten things. But at the very lease we should check for this case an throw and exception.

-----

Let me know if this is what you had in mind, I was also not sure how to verify return type consistency here.